src/gluonts/model/deepar/_estimator.py

```diff
@@ -238,7 +250,16 @@ def __init__(
             else DummyValueImputation(self.distr_output.value_in_support)
         )
 
-        self.train_sampler = train_sampler
+        self.train_sampler = (
+            train_sampler
+            if train_sampler is not None
+            else ExpectedNumInstanceSampler(1.0, min_future=prediction_length)
```

Comment:
I guess we can't have these as default, since it depends on the `prediction_length`?

Reply:
That's correct. It's unfortunate...